### ISSUE TITLE
WriteColoredOutput vs Write-Error

### ISSUE THREAD

**Issue by dcorriveau-omniscient** [psake/psake#175]
Hey ! I'm using CCNet as my buils server, and from what I can tell, they are listening on the error stream to determine errors (basically all it does is it sets the `<message level="Error">` tag so that it can be parsed and colored appropriately on the dashboard. However, since psake is outputting catched errors using `WriteColoredOutput` (well, manually changing the color of the shell text), those error messages aren't displayed red in the dashboard.

Is there an option that I missed that would allow me to achieve what I want, or could I add the option myself to tell psake to use write-error instead ? Would you accept that pull request ? Thanks !

**Comment by mrhockeymonkey** [comment]
I agree with the above. In my mind since the function Invoke-PSake is using cmdletbinding() and therefore accepts -erroraction it should really be able to throw a real powershell error instead of output that is colored red. This has bred a bit of confusion at my place of work. I would be happy to submit such a pull request. Thoughts?

**Comment by UberDoodles** [comment]
This could be resolved to some extent by issue #65, for which I have raised a pull request (https://github.com/psake/psake/pull/301).

The changes in that pull request mean that all of psake's messages are categorised (e.g. "error", "warning", "success", etc), and you can override how each of those messages are routed. For backwards-compatibility, the messages are handled in the same way as they always have (using WriteColoredOutput).

If we were to change the code to use Write-Error instead of WriteColoredOutput, this would be a breaking change that could cause people a lot of pain. So it might be better to leave WriteColoredOutput as the default, but be able to override it using the #65 solution.

**Comment by HeyItsGilbert** [comment]
This should be resolved as of #301, which landed the ability to override how psake handles its internal logging. You can now intercept the message output (error / warning / success / etc.) and route errors to the actual error stream instead of relying on the colored host output that CI servers like CCNet can't parse. That covers the original ask here, so I'm closing this out. If the override hooks don't get you what you need for the CCNet case specifically, drop a comment and I'll reopen.
